juju/juju
# Add caas remotestate watcher, prepare for upgrade-charm

## Description of change

To support upgrade-charm on caas models, a few things need to happen. This PR is the first bit.
1. Stop the uniter from attempting to react to application changes on caas models.
2. Prepare for the caas operator to react to application changes.

Step 2 introduces a remotestate watcher which is a cut down version of what's used for uniter. We allow for > 1 subwatcher even though for now we just need the one. We can tidy up later if needed.

There's nothing new that's user visible here yet.

## QA steps

Deploy an IAAS model and verify that charms can be upgraded still.
Ensure a CAAS model can be deployed.
